projboard/tests/test_client.py

```diff
@@ -1,7 +1,34 @@
 import pytest
+from projboard.models.article import Article
+
+
+USERINPUT = 'userinput'
+ARTICLES = 'articles'
+COUNT = 'numArticles'
+VALUE = 'c'
 
 
 @pytest.mark.django_db
 def test_homepage(client):
     response = client.get("/")
     assert response.status_code == 200
+
+
+def test_empty_searchpage_results(client):
+    response = client.get("/search/")
+    assert response.status_code == 200
+    assert [] == response.context['articles']
+    assert "" == response.context['userinput']
+    assert 0 == response.context['numArticles']
+
+
+@pytest.mark.django_db
+def test_valued_searchpage_results(client, article):
+    response = client.post('/search/', {'title': VALUE})
+    assert response.status_code == 200
+    assert response.context[USERINPUT] == VALUE
+
+    for i in response.context[ARTICLES]:
+        assert i in Article.search_by_title(VALUE)
```

Comment:
I am not sure why there is a need to access to db,

You are running client test, consider populate what you need from the fixtures.
later compare the search results returned, against the fixtures content

BTW, only from tests, I can see that the following contexts returned
response.context[articles]
response.context[userinputs]
response.context[numarticles]

Consider renaming userinputs to better name such as searchtitle

Reply:
in test_article.py i have a func test_search_by_object that do search by title , is it the right way to call it and just pass the appropriate args to it ??

Reply:
it's not a fixture .

Reply:
@Yarboa i think that i can't use the fixures because when i use client fixure it uses the real database and the local one , i'm a littel bit confuse .

Reply:
You can have a fixture that queries 
Article.search_by_title(VALUE)
and hold the values, than you can compare it,

You are testing UI, test should not interact with DB
for that you can use fixtures, and their names handle what you need

Reply:
No change here
But resolving